apps/researcher/src/components/read-more-text.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+
+import {useState} from 'react';
+
+interface ReadMoreTextProps {
+  text?: string;
+  maxLength?: number;
+}
+
+export function ReadMoreText({text, maxLength = 300}: ReadMoreTextProps) {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  if (!text) {
+    return null;
+  }
+
+  if (text.length <= maxLength) {
+    return <p>{text}</p>;
+  }
+
+  return (
+    <p>
+      {isExpanded ? text : `${text.slice(0, maxLength)}... `}
```

Comment:
Instead of slicing, would it be possible to hide and show the full text with Tailwind's [truncate](https://tailwindcss.com/docs/text-overflow#truncate)?

Reply:
We used tailwinds `line-clamp` before; that works better for multi-line purposes. The problem is showing the "read-more" button conditionally. There is not a "if text is clamp" class.